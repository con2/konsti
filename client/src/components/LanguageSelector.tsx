import React, { FC, ReactElement, ChangeEvent } from 'react';
import { useTranslation } from 'react-i18next';
import { TFunction } from 'i18next';
import moment from 'moment';
import styled from 'styled-components';

export const LanguageSelector: FC = (): ReactElement => {
  const { t, i18n } = useTranslation();
  const language = i18n.language;

  // Language toggle
  const toggle = async (lng: string): Promise<TFunction> =>
    await i18n.changeLanguage(lng);

  const setLanguage = (event: ChangeEvent<HTMLSelectElement>): void => {
    toggle(event.target.value);
    moment.locale(event.target.value);
  };

  return (
    <>
      <LanguageSelectorContainer
        id='language'
        value={language}
        onChange={setLanguage}
      >
        <option title={t('language.english')} value='en'>
          {t('language.englishShort')}
        </option>
        <option title={t('language.finnish')} value='fi'>
          {t('language.finnishShort')}
        </option>
      </LanguageSelectorContainer>
    </>
  );
};

const LanguageSelectorContainer = styled.select`
  height: 30px;
  width: 60px;
`;
