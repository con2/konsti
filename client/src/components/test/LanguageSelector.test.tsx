import React from 'react';
import { shallow } from 'enzyme';
import { LanguageSelector } from '../LanguageSelector';

describe('LanguageSelector', () => {
  it('should render correctly', () => {
    const component = shallow(<LanguageSelector />);
    expect(component).toMatchSnapshot();
  });
});
