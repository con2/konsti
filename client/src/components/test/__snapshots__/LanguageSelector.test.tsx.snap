// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector should render correctly 1`] = `
<Fragment>
  <LanguageSelector__LanguageSelectorContainer
    id="language"
    onChange={[Function]}
    value="en"
  >
    <option
      title="language.english"
      value="en"
    >
      language.englishShort
    </option>
    <option
      title="language.finnish"
      value="fi"
    >
      language.finnishShort
    </option>
  </LanguageSelector__LanguageSelectorContainer>
</Fragment>
`;
